packages/components/src/util/typography.tsx

```diff
@@ -70,9 +68,32 @@ function Typography<IComponent extends React.ElementType>({
   const Component = as;
   return (
     <Component
-      className={cx(`typography--size-${size}`, `typography--color-${color}`, {
-        [`typography--weight-${weight}`]: weight,
-      })}
+      className={clsx(
+        styles.typography,
+        // Sizes
+        size === "h1" && styles.sizeH1,
+        size === "h2" && styles.sizeH2,
+        size === "h3" && styles.sizeH3,
+        size === "h4" && styles.sizeH4,
+        size === "h5" && styles.sizeH5,
+        size === "body" && styles.sizeBody,
+        size === "sm" && styles.sizeSm,
+        size === "xs" && styles.sizeXs,
+        size === "caption" && styles.sizeCaption,
+        size === "overline" && styles.sizeOverline,
+        // Colors
+        color === "alert" && styles.colorAlert,
+        color === "base" && styles.colorBase,
+        color === "brand" && styles.colorBrand,
+        color === "info" && styles.colorInfo,
+        color === "neutral" && styles.colorNeutral,
+        color === "success" && styles.colorSuccess,
+        color === "warning" && styles.colorWarning,
+        color === "white" && styles.colorWhite,
+        // Weights
+        weight === "bold" && styles.weightBold,
+        weight === "normal" && styles.weightNormal,
```

Comment:
Mmm too bad you have to write everything out like this. But the string interpolation is easier to mess up, and I don't think it throws an error if you do; the class just isn't applied.

Reply:
I was referring to mistyping the class name which creates a legitimate string but it's not associated with an actual style (I did this yesterday 😅).

Reply:
good points abt the downsides of string interpolation! I think this component may be the most cases to type out, and it's not terrible.

I wonder if it's worthwhile to break it up some (for readability) -- e.g. moving the logic of picking a className for size to a `sizeStyle` function? could do the same for button, though that one's not as bad

Reply:
I wanted to keep it in one place to see:
1. how extreme it would get
2. all the possible classes in one place
3. an extreme case we could share as a possible strategy to product engineers

> I think this component may be the most cases to type out, and it's not terrible.
I agree here, and I think this style, while not always beautiful, leads to really clear direction on what to do

Reply:
> I don't think it throws an error if you do; the class just isn't applied.

I don't think this throws an error either, since it gets the following types:
```ts
declare module "*.module.css" {
  const content: Record<string, string>;
  export default content;
}
```

We could explore something like https://github.com/mrmckeb/typescript-plugin-css-modules to ensure the types work. There's also https://www.npmjs.com/package/css-modules-flow-types-cli on the flow side for the app.